generate/generate.go

```diff
@@ -904,11 +904,14 @@ func (g Generator) AddBindMount(bind string) error {
 }
 
 // SetupPrivileged sets up the priviledge-related fields inside g.spec.
-func (g Generator) SetupPrivileged(privileged bool) {
+func (g Generator) SetupPrivileged(privileged bool, hostSpecific bool) {
 	if privileged {
 		// Add all capabilities in privileged mode.
 		var finalCapList []string
 		for _, cap := range capability.List() {
+			if hostSpecific && cap > capability.CAP_LAST_CAP {
+				continue
+			}
 			finalCapList = append(finalCapList, fmt.Sprintf("CAP_%s", strings.ToUpper(cap.String())))
```

Comment:
On Thu, Jul 21, 2016 at 03:39:31PM -0700, Mrunal Patel wrote:
> >  		// Add all capabilities in privileged mode.
> >  		var finalCapList []string
> >  		for _, cap := range capability.List() {
> > +			if hostSpecific && cap > capability.CAP_LAST_CAP {
> > +				continue
> > +			}
> >  			finalCapList = append(finalCapList, fmt.Sprintf("CAP_%s", strings.ToUpper(cap.String())))
> 
> This could be addressed in a different PR.

Using AddProcessCapability here?  I guess, if we can't decide in this
one.  I'm happy to file that PR if you want to land this without
addressing that point (although I still think we want to handle [1]
here).

[1]: https://github.com/opencontainers/ocitools/pull/138/files#r71790239